xlators/features/trash/src/trash.c

```diff
@@ -1974,16 +1975,17 @@ trash_truncate(call_frame_t *frame, xlator_t *this, loc_t *loc, off_t offset,
         goto out;
     }
     /* This will be more accurate */
-    inode_path(loc->inode, NULL, &pathbuf);
+    retval = inode_path(loc->inode, NULL, &pathbuf);
 
     /* Checks whether file is in trash directory or eliminate path.
      * In all such cases it does not move to trash directory,
      * truncate will be performed
      */
     match = check_whether_eliminate_path(priv->eliminate, pathbuf);
```

Comment:
`pathbuf` is used inside this function, so we need to check `retval` before calling it.

Reply:
added it and also removed pathbuf==NULL check from check_whether_eliminate_path function.